### PR TITLE
[Node] Create sample application

### PR DESCRIPTION
**Pull request by majanjua-amzn**
## Issue
Starting development for aws-otel-js-instrumentation's enablement canaries and release tests. These require a sample application with certain specifications written in some NodeJS framework.

Note: This PR is a preliminary PR to put up most of the functionality and the sample app deployment scripts. Fixes will likely be made once the tests are written to ensure, for example, async behaviour is done appropriately and is generating the correct telemetry.

## Description of changes
- Created NodeJS/ExpressJS application with all test APIs implemented
- Created Dockerfiles and docker-compose.yml to generate docker images for the sample application (frontend and remote)
- Created workflows to generate ECR images and ZIPs pushed to S3
  - Renamed all sample app deployment files: They will now appear in a much shorter, more readable and more organized format in the Actions page of this repo.

## Rollback procedure
Can just revert and redeploy sample apps for languages other than Node.
Node specifically will not need revert since we don't test it yet.

## Testing
- [ECR Deployment](https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/10582013619)
- [S3 Deployment](https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/10582013624)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sample-apps/node/frontend-service/index.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    rs.setEncoding('utf8');
+    rs.on('data', (result) => {
+      res.send(`/client-call response: ${result}`);
+    });
+  });
+  request.on('error', (err) => {});
+  request.end();
+});
+
+app.get('/mysql', (req, res) => {
```
Comment (pxaws) [inline sample-apps/node/frontend-service/index.js:95]: Any reason why we want to choose mysql?

File: .github/workflows/node-sample-app-ecr-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.E2E_IAD_TEST_ACCOUNT_ARN }}
```
Comment (harrryr) [inline .github/workflows/node-sample-app-ecr-deploy.yml:30]: Can we update this to `arn:aws:iam::${{ env.E2E_TEST_ACCOUNT_ID }}:role/${{ env.E2E_TEST_ROLE_NAME }}`

File: .github/workflows/node-sample-app-ecr-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+      - name: Docker compose build
+        working-directory: sample-apps/node
+        run: |
+          docker compose build
+
+      - name: Upload Main Service Image
+        run: |
+          docker tag nodesampleapp/frontend-service ${{ env.ACCOUNT_ID }}.dkr.ecr.${{ matrix.aws-region }}.amazonaws.com/${{ secrets.APP_SIGNALS_NODE_E2E_FE_SA_IMG }}
+          docker push ${{ env.ACCOUNT_ID }}.dkr.ecr.${{ matrix.aws-region }}.amazonaws.com/${{ secrets.APP_SIGNALS_NODE_E2E_FE_SA_IMG }}
```
Comment (harrryr) [inline .github/workflows/node-sample-app-ecr-deploy.yml:57]: ${{ secrets.APP_SIGNALS_NODE_E2E_FE_SA_IMG }}

Add this value to secret manager and retrieve the value through `aws-secretsmanager-get-secrets` action

File: .github/workflows/node-sample-app-ecr-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      
+      - name: Upload Remote Service Image
+        run: |
+          docker tag nodesampleapp/remote-service ${{ env.ACCOUNT_ID }}.dkr.ecr.${{ matrix.aws-region }}.amazonaws.com/${{ secrets.APP_SIGNALS_NODE_E2E_RE_SA_IMG }}
+          docker push ${{ env.ACCOUNT_ID }}.dkr.ecr.${{ matrix.aws-region }}.amazonaws.com/${{ secrets.APP_SIGNALS_NODE_E2E_RE_SA_IMG }}
```
Comment (harrryr) [inline .github/workflows/node-sample-app-ecr-deploy.yml:62]: Same here

File: .github/workflows/node-sample-app-s3-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.E2E_IAD_TEST_ACCOUNT_ARN }}
```
Comment (harrryr) [inline .github/workflows/node-sample-app-s3-deploy.yml:30]: Same as above

File: sample-apps/node/frontend-service/index.js (hunk 1 above)
Comment (majanjua-amzn) [reply sample-apps/node/frontend-service/index.js:95]: To cover the RDS DB use case, we can cover the MongoDB use case separately

File: .github/workflows/node-sample-app-ecr-deploy.yml (hunk 3 above)
Comment (majanjua-amzn) [reply .github/workflows/node-sample-app-ecr-deploy.yml:64]: Not done for other languages, if we're going to do this we should automate it for all and do it at once. Me doing it here would be throwaway effort.

**Review by harrryr — APPROVED** [review]

**Pull request merged**